src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2286,48 +2286,86 @@ private IOperation CreateBoundDiscardPatternOperation(BoundDiscardPattern boundN
+                var reference = OperationFactory.CreateInvalidOperation(_semanticModel, subpatternSyntax, ImmutableArray<IOperation>.Empty, isImplicit: true);
+                return new PropertySubpatternOperation(reference, pattern, _semanticModel, subpatternSyntax, isImplicit: false);
+            }
 
-            switch (member.Symbol)
+            ITypeSymbol previousType = matchedType;
+            do
             {
-                case FieldSymbol field:
-                    {
-                        var constantValue = field.GetConstantValue(ConstantFieldsInProgress.Empty, earlyDecodingWellKnownAttributes: false);
-                        return new FieldReferenceOperation(field.GetPublicSymbol(), isDeclaration: false,
-                            // PROTOTYPE(extended-property-patterns) We shouldn't use StrippedType here, maybe only if this is a nested member?
-                            createReceiver(), _semanticModel, member.Syntax, field.Type.StrippedType().GetPublicSymbol(), constantValue, isImplicit: false);
-                    }
-                case PropertySymbol property:
-                    {
-                        return new PropertyReferenceOperation(property.GetPublicSymbol(), ImmutableArray<IArgumentOperation>.Empty,
-                            // PROTOTYPE(extended-property-patterns) We shouldn't use StrippedType here, maybe only if this is a nested member?
-                            createReceiver(), _semanticModel, member.Syntax, property.Type.StrippedType().GetPublicSymbol(), isImplicit: false);
-                    }
-                default:
-                    {
-                        // We should expose the symbol in this case somehow:
-                        // https://github.com/dotnet/roslyn/issues/33175
-                        IOperation? receiver = CreatePropertySubpatternMember(member.Receiver, matchedType);
-                        var children = receiver is null ? ImmutableArray<IOperation>.Empty : ImmutableArray.Create(receiver);
-                        return OperationFactory.CreateInvalidOperation(_semanticModel, member.Syntax, children, isImplicit: false);
-                    }
+                var nameSyntax = member.Syntax;
+                var inputType = member.Receiver?.Type.StrippedType().GetPublicSymbol() ?? matchedType;
+                if (result is null)
+                {
+                    // Create an operation for last property access:
+                    // `{ SingleProp: <pattern operation> }`
+                    // or
+                    // `.LastProp: <pattern operation>` portion (treated as `{ LastProp: <pattern operation> }`)
+                    IPatternOperation pattern = (IPatternOperation)Create(subpattern.Pattern);
+                    result = createPropertySubpattern(member.Symbol, pattern, inputType, nameSyntax, isSingle: member.Receiver is null);
+                }
```

Comment:
Consider moving this case before the `do { ... } while (...);` loop. It doesn't look like it shares anything with the rest of the loop.